scripts/troubleshoot/collect_logs.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+
+# This script pulls logs from the replicaset agent pod and a random daemonset pod. This script is to make troubleshooting faster
+
+export ds_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-[a-z0-9]{5} | head -n 1)
+export ds_win_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-win-[a-z0-9]{5} | head -n 1)
+export rs_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-rs-[a-z0-9]{5} | head -n 1)
+
+echo "collecting logs from ${ds_pod}, ${ds_win_pod}, and ${rs_pod}"
+echo "    note: some erros are expected for clusters without windows nodes, they can safely be disregarded (filespec must match the canonical format:, zip warning: name not matched: omsagent-win-daemonset-fbit)"
+
+kubectl cp ${ds_pod}:/var/opt/microsoft/docker-cimprov/log omsagent-daemonset --namespace=kube-system --container omsagent
```

Comment:
Thanks David. also would be good to add kubectl describe pod for each and dump to a file. For liveness failures, it will tell what caused them, and also how frequently its happening.

Reply:
Good idea, will add